lib/object/update-in.ts

```diff
@@ -1,34 +1,12 @@
-// @ts-ignore
-import { get } from "../list/get";
-import { cloneDeep } from "../generic/clone-deep";
-
-/**
- * @param {{}} obj
- * @param {string[]} keys
- * @param {Function} fn
- * @param {*} rest
- * @returns {*}
- */
-export function updateIn(obj: Object, keys: any[], fn: Function, ...rest: any) {
-  const propertyPath = keys.join(".");
-  const updateValue = get(obj, propertyPath);
-  const result = fn(updateValue, ...rest);
-
-  const updatedObject = cloneDeep(obj);
-
-  function updateProps(obj: Object, keys: string[]) {
-    const next = keys[1];
-    const cur = keys[0];
-    if (!next) {
-      // @ts-ignore
-      obj[cur] = result;
-    } else {
-      // @ts-ignore
-      updateProps(obj[cur], keys.slice(1, keys.length));
-    }
+// @ts-nocheck
+import { next } from "../list/next";
+
+export function updateIn(map: object, keys: string[], func: Function, ...args: any): any {
+  let current = keys[0];
+  let rest = next(keys);
+  if (rest) {
+    return Object.assign(map, { [current]: updateIn(map[current], rest, func, ...args) });
```

Comment:
updateIn docs:
"Updates" a value in a nested associative structure, where ks is a sequence of keys and f is a function that will take the old value and any supplied args and return the new value, and returns a new nested structure. If any levels do not exist, hash-maps will be created.

This implementation will fail if map doesn't have a nested specified key.
For instance 
If we call 
```
const obj = {a: { b: 10 } };
updateIn(obj, ["a", "b", "c"], add, 10);
```
this function will fail with TypeError: Cannot read property 'c' of undefined

Please write a **test case** that will cover expected behaviour.